builder.py

```diff
@@ -0,0 +1,208 @@
+            b1a_decoder = tf.Variable(tf.random_normal([11 * sequence_length]), name='b1_away_decoder')
+            L1a_decoder = self.selu(tf.matmul(L1a_encoder, W1h_decoder) + b1h_decoder)
+            L1a_decoder = self.dropout_selu(L1h_decoder, keep_prob=self.keep_prob)
+
+            M = tf.concat([L1h_decoder, L1a_decoder], 1)
+
+            W2 = tf.get_variable("W2", shape=[11 * sequence_length * 2, 11 * sequence_length],
+                                 initializer=tf.contrib.layers.xavier_initializer())
+            b2 = tf.Variable(tf.random_normal([11 * sequence_length]), name='b2')
+            L2 = self.selu(tf.matmul(M, W2) + b2)
+            L2 = self.dropout_selu(L2, keep_prob=self.keep_prob)
+
+            W3 = tf.get_variable("W3", shape=[11 * sequence_length, int(11 * sequence_length / 2)],
+                                 initializer=tf.contrib.layers.xavier_initializer())
+            b3 = tf.Variable(tf.random_normal([int(11 * sequence_length / 2)]), name='b3')
+            L3 = self.selu(tf.matmul(L2, W3) + b3)
+            L3 = self.dropout_selu(L3, keep_prob=self.keep_prob)
+
+            W4 = tf.get_variable("W4", shape=[int(11 * sequence_length / 2), 2],
+                                 initializer=tf.contrib.layers.xavier_initializer())
+            b4 = tf.Variable(tf.random_normal([2]), name='b4')
+
+            # param_list = [W1h, W1a, b1h, b1a, W2, b2, W3, b3]
+            # self.saver = tf.train.Saver(param_list)
+
+            self.hypothesis = tf.matmul(L3, W4) + b4 # Probability of winning
+
+            tf.add_to_collection("logit", self.hypothesis)
+        # define cost/loss & optimizer
+        self.cost = tf.reduce_sum(tf.square(self.hypothesis - self.Y))
+        self.optimizer = tf.train.AdamOptimizer(learning_rate=learn_rate).minimize(self.cost)
+
+        # Test model and check accuracy
+
+    def get_accuracy(self, x_test_home, x_test_away, y_test, keep_prop=1.0):
+        # compare hypothesis mapped to 1 and 0 depending on whether home team won or not
+        # with
+
+        predict = tf.map_fn(
+            lambda x: x[0] > x[1],
+            self.sess.run(
+                self.hypothesis, 
+                feed_dict={
+                self.X_home: x_test_home, 
+                self.X_away: x_test_away, 
+                self.Y: y_test, 
+                self.keep_prob: keep_prop}
+            ), 
+            dtype=bool)
+
+        real = tf.map_fn(
+            lambda x: x[0] > x[1],
+            y_test,
+            dtype=bool)
+
+        return self.sess.run(
+            tf.divide(
+                tf.reduce_sum(tf.cast(tf.equal(predict, real), dtype=tf.int32)), len(y_test)))
+
+    def train(self, x_test_home, x_test_away, y_data, keep_prop):
+        return self.sess.run([self.cost, self.optimizer], feed_dict={
+            self.X_home: x_test_home, self.X_away: x_test_away, self.Y: y_data, self.keep_prob: keep_prop})
+
+    def save(self):
+        if not os.path.exists(DIRNAME):
```

Comment:
not sure if this necessary? simply saving might create missing folders by default. could be wrong.